include/emitc/emitc_tosa.h

```diff
@@ -449,6 +449,43 @@ inline Dest reshape(Src x) {
   return emitc::reshape<Dest>(x);
 }
 
+// PadOp
+template <typename Dest, typename Src, typename Padding>
+inline Dest pad(Src operand, Padding padding) {
+  using ET_Padding = typename get_element_type<Padding>::type;
+
+  static_assert(is_tensor<Dest>::value, "Expected tensor result");
+  static_assert(is_tensor<Src>::value, "Expected tensor argument");
+  static_assert(is_tensor<Padding>::value, "Expected tensor argument");
+
+  static_assert(Padding::rank() == 2, "Padding must have rank 2");
+  static_assert(Padding::dim(0) == Src::rank(),
+                "Dimension 1 of padding must equal source rank");
+  static_assert(Padding::dim(1) == 2, "Dimension 2 of padding is must be 2");
+
+  // this check is not needed in a conversion pipeline since this would be
+  // already illegal IR. Might be helpful for unittests, ect.
```

Comment:
NIT:
```suggestion
  // This check is not needed in a conversion pipeline since this would be
  // already illegal IR. Might be helpful for unittests, etc.
```